### PR TITLE
feat: auto claim faucet for test accounts of testnet_v1

### PR DESCRIPTION
**Pull request by ShookLyngs**
### Overview
This PR should provide a scheduled workflow that will automatically calim faucet for test accounts every 24 hours.  
Right now the main feature (claim faucet and scheduled workflow) of the PR is already done, but there're something left that might need improvement, we should discuss.

### Faucet method
Right now the commands to claim faucet manually:
```shell
# Claim faucet to an ETH address
$ npm run faucet -- claim -e <ETH_ADDRESS>
```
The `claim` command is made for single account claiming, so in the auto claiming workflow:
```yml
run: |
  npm run faucet -- claim -p $KEY1
  npm run faucet -- claim -p $KEY2
  npm run faucet -- claim -p $KEY3
```
Should we find a better way to run commands in the workflow?

### Local test accounts
We have sevaral pairs of accounts for different situations:
- For general situation like running tests in local environment, we have open accounts for everyone
- For GitHub Actions, we have accounts that are stored in GitHub Actions secrets

Right now the auto claiming workflow only claim faucet for accounts stored in GitHub Actions secrets.
The question is, should we claim for those open accounts, too?

If yes, should we simply add those open accounts into the workflow?

### CODE REVIEW

**Comment by Flouse** [comment]
> should we simply add those open accounts into the workflow?

yes, please.

> Should we find a better way to run commands in the workflow?

I think it's good enough for now.

**Comment by ShookLyngs** [comment]
@Flouse Added hardcode in the workflow to also claim faucet for open accounts.

File: scripts/account-faucet/src/faucet/command.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ethAddress?: HexString | string;
+  rpc?: string;
+}
+
+export function setupFaucetCommand(program: Command) {
+  program
+    .command('claim')
+    .description('claim faucet for ckb account')
+    .option('-p, --private-key <privateKey>', '[HexString] ckb private key')
+    .option('-c --ckb-address <ckbAddress>', '[HexString] ckb address (optional)')
```
Comment (Flouse) [inline scripts/account-faucet/src/faucet/command.ts]: Maybe we could use the this [address](https://pudge.explorer.nervos.org/address/ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x) as `ckbAddress` by default?

Comment (ShookLyngs) [reply scripts/account-faucet/src/faucet/command.ts]: > Maybe we could use the this [address](https://pudge.explorer.nervos.org/address/ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x) as `ckbAddress` by default?

Why this specific address? 

Condition checks of this part:
- passing a single argument `privateKey`
- or passing `ckbAddress` and `ethAddress` at the same time

Comment (Flouse) [reply scripts/account-faucet/src/faucet/command.ts]: > Why this specific address?

This address is the sender of [CKB Faucet](https://faucet.nervos.org/).

Its lock could be used as the default ownerLock:
```js
  # https://pudge.explorer.nervos.org/address/ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x
  const defaultOwnerLock: Script = {
    code_hash: '0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8',
    hash_type: 'type',
    args: '0x3f1573b44218d4c12a91919a58a863be415a2bc3'
  };
```

**Comment by ShookLyngs** [comment]
In latest commits:
- `ckbAddress` is now optional, you can pass `ckbPrivateKey` or `ethAddress` to claim faucet
- Improved submitted log, now the exact claim amount can be printed

File: scripts/account-faucet/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+You can pass a CKB private key, the CLI will automatically calculate the layer 2 deposit address, and then claim faucet to the address. Or you can pass `CKB address` and `ETH address` pair, the CLI will do the same thing for you.
+
+## Usage
+Install dependencies
+```bash
+$ npm install
+```
+Claim faucet, using your `CKB Layer 1 Private Key`
+```bash
+$ npm run faucet -- -p <CKB_PRIVATE_KEY>
```
Comment (Flouse) [inline scripts/account-faucet/README.md]:
```suggestion
$ npm run faucet -- claim -p <CKB_PRIVATE_KEY>
```

File: scripts/account-faucet/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+```
+Claim faucet, using your `ETH Address`
+```bash
+$ npm run faucet -- -e <ETH_ADDRESS>
```
Comment (Flouse) [inline scripts/account-faucet/README.md]:
```suggestion
$ npm run faucet -- claim -e <ETH_ADDRESS>
```

File: scripts/account-faucet/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+```
+When Claiming with `ETH Address`, you can also pass your `CKB Address`
+```bash
+$ npm run faucet -- -e <ETH_ADDRESS> - c <CKB_ADDRESS>
```
Comment (Flouse) [inline scripts/account-faucet/README.md]:
```suggestion
$ npm run faucet -- claim -e <ETH_ADDRESS> - c <CKB_ADDRESS>
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: scripts/account-faucet/src/faucet/command.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import config from '../config';
+import { Command } from 'commander';
+import { Address, HexString } from '@ckb-lumos/base';
+import { GodwokenWeb3 } from '../godwoken/web3';
+import { claimFaucetForCkbAddress } from './faucet';
+import { encodeLayer2DepositAddress, privateKeyToLayer2DepositAddress, toAddress, toHexString } from './address';
+
+const defaultCkbAddress = '0xckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x';
```
Comment (Flouse) [inline scripts/account-faucet/src/faucet/command.ts]:
```suggestion
const defaultCkbAddress = 'ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x';
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: scripts/account-faucet/src/faucet/command.ts (hunk 5 above)
Comment (Flouse) [reply scripts/account-faucet/src/faucet/command.ts]: It's not hex format.

**Review by Flouse — APPROVED** [review]

**Pull request merged**